Model/Api/UniversalApi.php

```diff
@@ -0,0 +1,250 @@
+        $this->bugsnag = $bugsnag;
+        $this->logHelper = $logHelper;
+        $this->errorResponse = $errorResponse;
+        $this->response = $response;
+    }
+
+    public function execute(
+        $event = null,
+        $data = null
+    )
+    {
+        try {
+            switch($event){
+                case "order.create":
+                    //currently sends its own response updated for v2, no return
+                    $this->createOrder->execute(
+                        $event,
+                        isset($data['order']) ? $data['order'] : null,
+                        isset($data['currency']) ? $data['currency'] : null
+                    );
+                    break;
+                case "cart.create":
+                    //sends response itself, updated for v2
+                    $this->orderManagement->createCart(
+                        isset($data['items']) ? $data['items'] : null,
+                        isset($data['currency']) ? $data['currency'] : null
+                    );
+                    break;
+                case "discounts.code.apply":
+                    //sends response itself
+                    $this->updateCart->discountHandler(
+                        isset($data['discount_code']) ? $data['discount_code'] : null,
+                        isset($data['cart']) ? $data['cart'] : null,
+                        isset($data['customer_name']) ? $data['customer_name'] : null,
+                        isset($data['customer_email']) ? $data['customer_email'] : null,
+                        isset($data['customer_phone']) ? $data['customer_phone'] : null
+                    );
+                    break;
+                case "cart.update":
+                    //tried to get this to return a result but the M2 dataobject processor didn't want to play
+                    //nice, so this will also just send a response itself until MA-483
+                    $updateResult = $this->updateCart->execute(
+                        isset($data['cart']) ? $data['cart'] : null,
+                        isset($data['add_items']) ? $data['add_items'] : null,
+                        isset($data['remove_items']) ? $data['remove_items'] : null,
+                        isset($data['discount_codes_to_add']) ? $data['discount_codes_to_add'] : null,
+                        isset($data['discount_codes_to_remove']) ? $data['discount_codes_to_remove'] : null
+                    );
+                    break;
+                case "order.shipping_and_tax":
+                    //Returns ShippingOptionsInterface
+                    $this->result->setData(
+                        $this->shippingMethods->getShippingMethods(
+                            isset($data['cart']) ? $data['cart'] : null,
+                            isset($data['shipping_address']) ? $data['shipping_address'] : null
+                        )
+                    );
+                    break;
+                case "order.shipping":
```

Comment:
Could you explain why we have differenct code for "order.shipping_and_tax" and "order.shipping"?

Reply:
The [merchant API docs](https://docs.bolt.com/api_reference#tag/Merchant-API) specify different payloads and responses. It's probably possible to streamline this, as well as the "order.tax", but for now I was simply re-using the existing handlers as much as I could.